cmake/polysolve/polysolve_target_link_system_libraries.cmake

```diff
@@ -0,0 +1,28 @@
+function(polysolve_target_link_system_libraries target)
```

Comment:
What is the point of this? For hiding third-party warnings, I would recommend using a build-system agnostic solution, using #pragma via a [warnoff](https://github.com/adobe/lagrange/blob/main/modules/core/include/lagrange/utils/warnoff.h)/[warnon](https://github.com/adobe/lagrange/blob/main/modules/core/include/lagrange/utils/warnon.h) mechanism. This avoids fiddling with CMake commands...